examples/minimal-mdns/tester.cpp

```diff
@@ -0,0 +1,303 @@
+
+using namespace chip::ArgParser;
+
+constexpr uint16_t kOptionType = 't';
+
+// non-ascii options have no short option version
+constexpr uint16_t kOptionListenPort = 0x100;
+constexpr uint16_t kOptionRuntimeMs  = 0x102;
+constexpr uint16_t kOptionTraceTo    = 0x104;
+
+// Only used for argument parsing. Tracing setup owned by the main loop.
+chip::CommandLineApp::TracingSetup * tracing_setup_for_argparse = nullptr;
+
+bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier, const char * aName, const char * aValue)
+{
+    switch (aIdentifier)
+    {
+    case kOptionListenPort:
+        if (!ParseInt(aValue, gOptions.listenPort))
+        {
+            PrintArgError("%s: invalid value for listen port: %s\n", aProgram, aValue);
+            return false;
+        }
+        return true;
+    case kOptionTraceTo:
+        tracing_setup_for_argparse->EnableTracingFor(aValue);
+        return true;
+    case kOptionType:
+        if (strcasecmp(aValue, "SRV") == 0)
+        {
+            gOptions.type = TestType::kSrv;
+        }
+        else
+        {
+            PrintArgError("%s: invalid value for query type: %s\n", aProgram, aValue);
+            return false;
+        }
+        return true;
+
+    case kOptionRuntimeMs:
+        if (!ParseInt(aValue, gOptions.runtimeMs))
+        {
+            PrintArgError("%s: invalid value for runtime ms: %s\n", aProgram, aValue);
+            return false;
+        }
+        return true;
+    default:
+        PrintArgError("%s: INTERNAL ERROR: Unhandled option: %s\n", aProgram, aName);
+        return false;
+    }
+}
+
+OptionDef cmdLineOptionsDef[] = {
+    { "listen-port", kArgumentRequired, kOptionListenPort },
+    { "type", kArgumentRequired, kOptionType },
+    { "timeout-ms", kArgumentRequired, kOptionRuntimeMs },
+    { "trace-to", kArgumentRequired, kOptionTraceTo },
+    {},
+};
+
+OptionSet cmdLineOptions = { HandleOptions, cmdLineOptionsDef, "PROGRAM OPTIONS",
+                             "  --listen-port <number>\n"
+                             "        The port number to listen on\n"
+                             "  -t\n"
+                             "  --type\n"
+                             "        The packet to test with: \n"
+                             "  --timeout-ms\n"
+                             "        How long to wait for replies\n"
+                             "  --trace-to <dest>\n"
+                             "        trace to the given destination (supported: " SUPPORTED_COMMAND_LINE_TRACING_TARGETS ").\n"
+                             "\n" };
+
+HelpOptions helpOptions("minimal-mdns-tester", "Usage: minimal-mdns-tester [options]", "1.0");
+
+OptionSet * allOptions[] = { &cmdLineOptions, &helpOptions, nullptr };
+
+class ReportDelegate : public mdns::Minimal::ServerDelegate
+{
+public:
+    void OnQuery(const mdns::Minimal::BytesRange & data, const chip::Inet::IPPacketInfo * info) override
+    {
+        char addr[32];
```

Comment:
This is not always enough. There is `IPAddress::kMaxStringLength`. Also, I think +1 is needed here. Same for all address buffers.

Reply:
This was a copy & paste. Adjusted all 3 example files (tester, client and server). Comments say it includes space for NUL so I did not add +1.